File: LAB6_MidtermLab/BookManager/backend/api/sales.py
# backend/api/sales
from fastapi import APIRouter, Depends, Body
from backend.crud import sell_book
from backend.schemas import BillRecordResponse
from backend.api.deps import get_db, get_current_user
from sqlalchemy.orm import Session
from backend.database.models import User

router = APIRouter(prefix="/sales", tags=["图书销售"])

@router.post("/{bk_id}", response_model=BillRecordResponse)
def sell_book_1(
    bk_id: int,
    quantity: int,
    db: Session = Depends(get_db),
    user: User = Depends(get_current_user)
) -> BillRecordResponse:
    return sell_book(db=db, book_id=bk_id, quantity_delta=quantity, customer=user)

File: LAB6_MidtermLab/BookManager/frontend/src/api/sales.js
// sales.js - 销售相关接口（销售图书）
import api from './api';

/**
 * 销售图书
 * @param {number} bk_id - 图书ID
 * @param {number} quantity - 销售数量
 * @returns {Promise} 返回销售记录的Promise对象
 */

export const sellBook = (bk_id, quantity) => {
  // 确保使用正确的路径前缀 '/sales'
  return api.post(`/sales/${bk_id}?quantity=${quantity}`);
};